### PR TITLE
refactor: add some bounds to improve type infer

### PR DESCRIPTION
**Pull request by M-Adoo**

### CODE REVIEW

File: src/observable.rs [hunk 1]
```diff
@@ -146,10 +146,10 @@ pub trait Observable {
   /// Creates a new stream which calls a closure on each element and uses
   /// its return as the value.
   #[inline]
-  fn map<B, Item, F>(self, f: F) -> MapOp<Self, F>
+  fn map<B, F>(self, f: F) -> MapOp<Self, F>
```
Comment (Mnwa) [inline src/observable.rs]: Maybe to write these changes to "breaking changes"? Because it can break some code with type explicit

**Comment by M-Adoo** [comment]
@Mergifyio rebase

**Comment by M-Adoo** [comment]
@Mergifyio update

**Comment by M-Adoo** [comment]
@Mnwa  `@Mergifyio update` work, but `@Mergifyio rebase` failure. And bot not merge this pr. I'll try to fix it in today or next. There may be some noise that will make you bored. Sorry.

**Comment by Mnwa** [comment]
No problem! Just ping me if you'll need next review :)

**Comment by M-Adoo** [comment]
@mergifyio rebase

**Comment by M-Adoo** [comment]
@Mergifyio rebase

**Comment by M-Adoo** [comment]
@Mnwa great! Mergify work now,  5 rules add in this project, see more detail in https://github.com/rxRust/rxRust/pull/89/checks?check_run_id=487964335.  You can call @Mergifyio to rebase or update branch before merge. After a core developers approve, pr'll auto merge.

**Comment by M-Adoo** [comment]
Please approve this pr again.

**Comment by Mnwa** [comment]
Auto mr did not work Auto mr did not work

**Comment by M-Adoo** [comment]
seems  status-success not trigger
image

**Comment by M-Adoo** [comment]
@Mnwa It's should worm,  you can try call mergifyio rebase  and approve it again.

**Comment by M-Adoo** [comment]
try call mergify rebase before approve
image

**Comment by Mnwa** [comment]
@Mergifyio rebase

**Comment by Mnwa** [comment]
Nice

**Pull request merged**